broker-daemon/utils/migrate-store.spec.js

```diff
@@ -1,117 +1,192 @@
-      { key: '3', value: '3rd value' }
-    ]
-    fakeOperations = [
-      { type: 'put', key: 'a', value: '1' },
-      { type: 'put', key: 'b', value: '2' },
-      { type: 'put', key: 'c', value: '3' }
-    ]
-
-    stream.on.callsFake((evt, fn) => {
-      if (evt === 'error') return
-      const data = evt === 'data' ? fn : function () {}
-      const end = evt === 'end' ? fn : function () {}
-      const records = fakeRecords.slice()
-
-      const nextRecord = () => {
-        if (records.length) {
-          data(records.shift())
-          process.nextTick(nextRecord)
-        } else {
-          end()
-        }
-      }
-
-      process.nextTick(nextRecord)
-    })
     createDbOperation = sinon.stub()
-    fakeRecords.forEach(({ key, value }, i) => {
-      createDbOperation.withArgs(key, value).returns(fakeOperations[i])
-    })
+    batchSize = 1
+    promisifyStub = sinon.stub().withArgs(targetStore.batch).returns(batchStub)
+
+    reverts = []
+    reverts.push(migrateStore.__set__('promisify', promisifyStub))
   })
 
-  it('returns a promise', () => {
-    expect(migrateStore(sourceStore, targetStore, createDbOperation)).to.be.a('promise')
+  afterEach(() => {
+    reverts.forEach(r => r())
   })
 
-  it('creates a readstream from the store', () => {
-    migrateStore(sourceStore, targetStore, createDbOperation)
+  it('returns a promise', () => {
+    expect(migrateStore(sourceStore, targetStore, createDbOperation, batchSize)).to.be.a('promise')
+  })
 
+  it('creates a read string from the source sublevel store', async () => {
+    migrateStore(sourceStore, targetStore, createDbOperation, batchSize)
+    expect(sourceStore.createReadStream).to.have.been.called()
     expect(sourceStore.createReadStream).to.have.been.calledOnce()
   })
 
-  it('sets an error handler', () => {
-    migrateStore(sourceStore, targetStore, createDbOperation)
+  it('migrates records to a target store', async () => {
+    const record = { key: '1', value: '1234' }
+    const batchEntry = { type: 'del', key: record.key }
+    onStub.withArgs('data', sinon.match.any).yields(record)
+    createDbOperation.withArgs(record.key, record.value).returns(batchEntry)
+    batchStub.resolves()
 
-    expect(stream.on).to.have.been.calledWith('error', sinon.match.func)
-  })
+    migrateStore(sourceStore, targetStore, createDbOperation, batchSize)
 
-  it('sets an end handler', () => {
-    migrateStore(sourceStore, targetStore, createDbOperation)
+    expect(batchStub).to.have.been.calledWith([batchEntry])
+    expect(pauseStub).to.have.been.calledBefore(batchStub)
+    expect(batchStub).to.have.been.calledBefore(resumeStub)
+  })
 
-    expect(stream.on).to.have.been.calledWith('end', sinon.match.func)
+  context('on error', () => {
+    it('rejects the promise if there was a failure', () => {
+      const promise = migrateStore(sourceStore, targetStore, createDbOperation, batchSize)
+      const error = onStub.args[2][1]
```

Comment:
[nit] would be better to use the name of the arg rather than call order to get the error handler.